kedro-training/docs/01_prerequisites.md

```diff
@@ -0,0 +1,77 @@
+
+### `PySpark`
+
+[Java](https://www.oracle.com/technetwork/java/javase/downloads/index.html) will need to be installed if `PySpark` is a workflow requirement.
+
+## Operating system specifications
+
+
+### Windows
+
+Windows users will require admin rights to install [Anaconda](https://www.anaconda.com/download) and [Java](https://www.oracle.com/technetwork/java/javase/downloads/index.html).
+
+### Build tools
+
+On Unix-like operating systems, you will need to install a C compiler and related build tools for your platform. This is due to the inclusion of the [memory-profiler](https://pypi.org/project/memory-profiler/) library in our dependencies. If your operating system is not mentioned, please refer to its documentation.
+
+#### macOS
+To install Command Line Tools for Xcode, run the following from the terminal:
+
+```bash
+xcode-select --install
+```
+
+#### GNU / Linux
+
+##### Debian / Ubuntu
+
+The following command (run with root permissions) will install the `build-essential` metapackage for Debian-based distributions:
+
+```bash
+apt-get update && apt-get install build-essential
+```
+
+##### Red Hat Enterprise Linux / Centos
+The following command (run with root permissions) will install the "Develop Tools" group of packages on RHEL/Centos:
+
+```bash
+yum groupinstall 'Development Tools'
+```
+
+## Optional tools
+
+### Git
+Git is a version control software that records changes to a file or set of files. Git is especially helpful for software developers as it allows changes to be tracked (including who and when) when working on a project.
+
+To download `git`, go to the following link and choose the correct version for your operating system: https://git-scm.com/downloads.
+
+#### Windows
+Download the [`git` for Windows installer](https://gitforwindows.org/). Make sure to select **Use `git` from the Windows command prompt** this will ensure that `git` is permanently added to your PATH.
+
+Also select **Checkout Windows-style, commit Unix-style line endings** selected and click on **Next**.
+
+This will provide you both `git` and `git bash`. We might have a few exercises using the command line quite a lot during the workshop so using `git bash` is a good option.
+
+### GitHub
+GitHub is a web-based service for version control using Git. You will need to set up an account at: https://github.com.
+
+Basic GitHub accounts are free and you can now also have private repositories.
+
+### Text editors & IDEs
+Text editors are tools with powerful features designed to optimize writing code. There are many text editors that you can choose from. Here are some we recommend:
+
+- [PyCharm](https://www.jetbrains.com/pycharm/download/)
+- [VS Code](https://code.visualstudio.com/)
+- [Atom](https://atom.io/)
+
+### Docker
```

Comment:
I feel like the below wording is a bit ambiguous whether they need to install Docker or not before the bootcamp. If we are sending this to them today, we can explicitly ask them to install Docker?

Reply:
As discussed offline - let's clearly mark this section as optional. I will probably walk them through Kedro-Docker usage quickly, but let's not expect them to repeat that themselves

Reply:
I marked in optional tools so they can choose to install it or not to.